File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        negative = False
        if x < 0: 
            negative = True
            x *= -1
            
        strx = str(x)[::-1]
        
        if(negative):
            strx = "-" + strx
            
        if int(strx) >= 2**31 - 1 or int(strx) <= -(2**31):
            return 0
        else:
            return int(strx)
        
        

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        bool isNegative = false;
        long xl = x;
        if(x < 0){
            isNegative = true;
            xl *= -1;
        }
        
        std::string strX = std::to_string(xl);
        std::string newString = "";
        for(int i = strX.length() - 1; i >= 0; i--){
            newString += strX[i];
        }
        
        if(isNegative){
            newString = "-" + newString;
        }
        
        if(std::stol(newString) > std::pow(2, 31) + 1 || std::stol(newString) < std::pow(-2, 31)){
            return 0;
        }else{
            return std::stoi(newString);
        }
    }
};